samples/azure-quantum/entanglement-swap/ES-quantinuum-qsharp.ipynb

```diff
@@ -0,0 +1,1120 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Quantum Teleportation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "$\\providecommand{\\ket}[1]{\\left|#1\\right\\rangle}$\n",
+    "$\\providecommand{\\bra}[1]{\\left\\langle#1\\right|}$\n",
+    "\n",
+    "In this sample we will be looking at using quantum teleportation on Azure Quantum.\n",
+    "\n",
+    "To refresh our mind let us review how quantum teleportation works.\n",
+    "\n",
+    "We have two parties, Alice and Bob, they would like to share information, and they have an entangled state (usually the Bell state $\\frac{1}{\\sqrt{2}} (\\ket{00} + \\ket{11})$). \n",
+    "If Alice now wishes to transfer quantum information, she will entangle some payload with her qubit and measure her qubit and payload.She will transmit the results of the measurement to Bob, who can use them to reconstruct Alice's payload on his entangled qubit.\n",
+    "\n",
+    "In the following we code will implement this operation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/x-qsharp-data": "[]",
+      "text/html": [
+       "<ul></ul>"
+      ],
+      "text/plain": []
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "open Microsoft.Quantum.Diagnostics;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/x-qsharp-data": "[\"CreateBellState\",\"Send\",\"Receive\"]",
+      "text/html": [
+       "<ul><li>CreateBellState</li><li>Send</li><li>Receive</li></ul>"
+      ],
+      "text/plain": [
+       "CreateBellState, Send, Receive"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "operation CreateBellState(alice : Qubit, bob : Qubit) : Unit is Adj + Ctl {\n",
+    "    H(alice);\n",
+    "    CNOT(alice, bob);\n",
+    "}\n",
+    "\n",
+    "operation Send(alice : Qubit, payload: Qubit) : (Result, Result) {\n",
+    "    CNOT(payload, alice);\n",
+    "    H(payload);\n",
+    "    return (M(alice), M(payload));\n",
+    "}\n",
+    "\n",
+    "operation Receive(bob : Qubit, (cr_x : Result, cr_z : Result)) : Unit {\n",
+    "    if (cr_x == One) { X(bob); }\n",
+    "    if (cr_z == One) { Z(bob); }\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will now use Q# visualization tools to show the unitary responsible from creating the Bell state.\n",
+    "Printing unitary transformations implemented by operations is a great way to check whether your implementation is mathematically correct (for small operations)."
```

Comment:
This a great way to phrase it — the transformation is not the operation but is implemented by the operation!

Reply:
Will use it for the qiskit version